rust/routee-compass-powertrain/src/routee/speed_grade_model.rs

```diff
@@ -152,6 +144,8 @@ impl TryFrom<(Arc<SpeedGradeModelService>, &serde_json::Value)> for SpeedGradeMo
     ) -> Result<Self, Self::Error> {
         let (service, conf) = input;
 
+        let model_names: Vec<&String> = service.energy_model_library.keys().collect();
```

Comment:
if we move line 147 to line 181, we only build this vector if there is an error, instead of building it each time we process a query

Reply:
yep yep, that's a good point, change forthcoming